paymentsheet/src/main/java/com/stripe/android/paymentsheet/forms/FormViewModel.kt

```diff
@@ -42,12 +48,32 @@ internal class FormViewModel @Inject internal constructor(
         @Suppress("UNCHECKED_CAST")
         override fun <T : ViewModel> create(modelClass: Class<T>): T {
             return FormViewModel(
-                elements = formElements,
+                initialElements = formElements,
                 formArguments = formArguments,
             ) as T
         }
     }
 
+    private val linkInlineSignupMode = formArguments.linkHandler?.linkSignupMode?.stateIn(
+        scope = viewModelScope,
+        started = SharingStarted.Eagerly,
+        initialValue = null,
+    ) ?: stateFlowOf(null)
+
+    val elements = initialElements.map { element ->
+        if (element is LinkEmptyFormElement) {
```

Comment:
This is exactly the pattern that would shouldn't introduce!

This makes testing and reasoning about things way more complicated!